### PR TITLE
[FLINK-5358] add RowTypeInfo exctraction in TypeExtractor

### PR DESCRIPTION
**Pull request by tonycox**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java [hunk 1]
```diff
@@ -345,8 +346,22 @@ public CustomType cross(CustomType first, Integer second) throws Exception {
...
-	
 
+	@Test
+	public void testRow() {
+		Row row = new Row(2);
+		row.setField(0, "string");
+		row.setField(1, 15);
+		TypeInformation<Row> rowInfo = TypeExtractor.getForObject(row);
+		Assert.assertEquals(rowInfo.getClass(), RowTypeInfo.class);
+		Assert.assertEquals(2, rowInfo.getArity());
```
Comment (wuchong) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java]: Can we also test the nested type info?  such as:
```java
assertEquals(
    new RowTypeInfo(
        BasicTypeInfo.STRING_TYPE_INFO, 
        BasicTypeInfo.INT_TYPE_INFO), 
    rowInfo);
```

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java [hunk 2]
```diff
@@ -345,8 +346,22 @@ public CustomType cross(CustomType first, Integer second) throws Exception {
...
+
+		Row nullRow = new Row(2);
+		TypeInformation<Row> genericRowInfo = TypeExtractor.getForObject(nullRow);
+		Assert.assertEquals(genericRowInfo, new GenericTypeInfo<>(Row.class));
+		int arity = genericRowInfo.getArity();
+		genericRowInfo.hashCode();
```
Comment (wuchong) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java]: Why do we call `getArity()` and `hashCode()` here ?

**Comment by tonycox** [comment]
Hi @fhueske could you look at this PR?

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java [hunk 3]
```diff
@@ -345,8 +346,25 @@ public CustomType cross(CustomType first, Integer second) throws Exception {
 
 		Assert.assertFalse(TypeExtractor.getForClass(PojoWithNonPublicDefaultCtor.class) instanceof PojoTypeInfo);
 	}
-	
 
+	@Test
+	public void testRow() {
```
Comment (fhueske) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java]: Please add a check for a `Row` with a `null` field.

**Comment by fhueske** [comment]
Thanks @tonycox. The changes look good. I'd would add one more test case to cover the case of `Row` objects with `null` fields.

File: flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java (hunk 3 above)
Comment (tonycox) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java]: in different test method? 
there is already Row with a null field check

`Row nullRow = new Row(2);
TypeInformation<Row> genericRowInfo = TypeExtractor.getForObject(nullRow);
Assert.assertEquals(genericRowInfo, new GenericTypeInfo<>(Row.class));`

Comment (fhueske) [inline flink-core/src/test/java/org/apache/flink/api/java/typeutils/TypeExtractorTest.java]: Oh, I'm sorry. I overlooked that check.

**Comment by fhueske** [comment]
Thanks @tonycox. PR is good to merge.

**Comment by twalthr** [comment]
+1 to merge

**Comment by fhueske** [comment]
merging

**Pull request closed without merging**